extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -130,5 +140,39 @@ void build(CombinedIndexBuildItem index,
                 }
             }
         }
+
+        // manage @BsonProperty for the @ProjectionFor annotation
+        Map<String, String> propertyMapping = new HashMap<>();
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_PROJECTION_FOR)) {
+            Type targetClass = annotationInstance.value().asClass();
+            ClassInfo target = index.getIndex().getClassByName(targetClass.name());
+            for (FieldInfo fieldInfo : target.fields()) {
+                if (fieldInfo.hasAnnotation(DOTNAME_BSON_PROPERTY)) {
+                    AnnotationInstance bsonProperty = fieldInfo.annotation(DOTNAME_BSON_PROPERTY);
+                    propertyMapping.put(fieldInfo.name(), bsonProperty.value().asString());
```

Comment:
This appears to map properties by name, but this can't work if we have more than one entity with the same field names, no?

Reply:
Yes, silly bug :(

Reply:
fixed